docker_custodian/docker_autostop.py

```diff
@@ -45,11 +44,28 @@ def stop_container(client, id):
         log.warn("Error stopping %s: %s" % (id, ae))
 
 
+def build_matcher(opts):
+    if opts.prefix:
+        return build_container_matcher(opts.prefix)
+    if opts.image:
+        return build_image_matcher(opts.image)
+
+
 def build_container_matcher(prefixes):
-    def matcher(name):
-        return any(name.startswith(prefix) for prefix in prefixes)
+    def matcher(container):
+        return any(
+            container.get('Name').lstrip('/').startswith(prefix)
+            for prefix in prefixes
+        )
     return matcher
 
+def build_image_matcher(images):
+    def matcher(container):
+        return any(
+            container.get('Config',{}).get('Image').startswith(image)
```

Comment:
Doesn't this need to be `.get('Image', "")` so that startswith is always called on a string?

Reply:
Sure